### ISSUE TITLE
Commit associated with tag `cardano-ledger-conway-1.17.2.0` is gone

### ISSUE THREAD

**Issue by KtorZ** [IntersectMBO/cardano-ledger#4742]
<img width="837" alt="image" src="https://github.com/user-attachments/assets/a63a0753-8c3f-4878-a7f9-1916e9ab4c6e">

:'(

**Comment by lehins** [comment]
Commit associated with a git tag cannot disappear. The fact that release was made from a branch that diverged from master branch is totally normal for a patch release.
Here is the tag: https://github.com/IntersectMBO/cardano-ledger/tree/cardano-ledger-conway-1.17.2.0

Feel free to reopen the ticket if I misunderstood you concern.

**Comment by KtorZ** [comment]
I don't think you're getting the issue @lehins. It is no longer possible to checkout this commit or pull the tag altogether because the associate git object no longer exist. This is likely because the branch it was on was deleted, but the commits never merged back to `master` (but were likely cherry-picked). 

The consequence is that the tag doesn't even show up if you `git fetch` the remote, and you cannot checkout any of the commit from the branch (or the tag itself, since it becomes an orphan reference associated to no object).

**Comment by lehins** [comment]
> It is no longer possible to checkout this commit or pull the tag altogether because the associate git object no longer exist.

If this was the case our release process would break couple years ago.

I suspect you doing something wrong. Could you please submit instructions on how to reproduce this issue.

Here is my attempt just now:

```shell
lehins@localhost:~/tmp> git clone git@github.com:IntersectMBO/cardano-ledger.git
Cloning into 'cardano-ledger'...
remote: Enumerating objects: 178560, done.
remote: Counting objects: 100% (11535/11535), done.
remote: Compressing objects: 100% (4696/4696), done.
remote: Total 178560 (delta 6551), reused 8853 (delta 4727), pack-reused 167025 (from 1)
Receiving objects: 100% (178560/178560), 114.56 MiB | 1.66 MiB/s, done.
Resolving deltas: 100% (100210/100210), done.
lehins@localhost:~/tmp> cd cardano-ledger/
lehins@localhost:~/tmp/cardano-ledger> git checkout cardano-ledger-conway-1.17.2.0
Note: switching to 'cardano-ledger-conway-1.17.2.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 0268e812b Bump up the version to `cardano-ledger-conway-1.17.2.0`
lehins@localhost:~/tmp/cardano-ledger> 
```

**Comment by KtorZ** [comment]
Indeed, that's curious. Cloning the repository anew works fine. Somehow, I cannot fetch this tag from my existing repository, which I've been doing for more than 4 years now 🙃. 

Go figure...

**Comment by lehins** [comment]
@KtorZ You don't have to do a fresh clone the repo to get the tag or a commit, you can always fetch them this way:
```shell
$ git fetch origin cardano-ledger-conway-1.17.2.0
From github.com:IntersectMBO/cardano-ledger
 * tag                   cardano-ledger-conway-1.17.2.0 -> FETCH_HEAD
```
FYI, this approach I believe even works for some commits that do not belong to any branches or are unassociated with any tags, just because github might need to keep them around forever anyways, so changes are visible in UI (eg. closed PRs etc). However, don't quote me on that.

**Comment by KtorZ** [comment]
>  you can always fetch them this way:

Well, no. Not in this case, as this is exactly what I typically do. I honestly don't know what happened but it looks like my local git got corrupt somehow. I am a fairly well seasoned git user 🤓, so I usually manage to solve my problems alone. And here, it looked like a case of a tag being pushed without the associated commit (a tag being merely just a pointer / reference to an object; but the said object must be in the database for the reference to be valid). I can fetch 1.17.0.0 just fine, but somehow, 1.17.2.0 is completely missing. 

I did recall seing (and ignoring) a warning when I fetched about some remote revision `test` that was no fetchable, but it looked quite unrelated. So 🤷.
